becquerel/core/fitting.py

```diff
@@ -72,6 +72,68 @@ def expgauss(x, amp=1, mu=0, sigma=1.0, gamma=1.0):
     return amp * (gamma / 2) * np.exp(arg1) * scipy.special.erfc(arg2)
 
 
+def gauss_dbl_exp(
+    x: np.ndarray,
+    amp: float,
+    mu: float,
+    sigma: float,
+    ltail_ratio: float,
+    ltail_slope: float,
+    ltail_cutoff: float,
+    rtail_ratio: float,
+    rtail_slope: float,
+    rtail_cutoff: float,
+):
+    """A Gaussian with exponential tails added to either side of the peak. This
+    is an extension of the Gaussian with a low-side exponential tail described
+    by Namboodiri et al here: https://www.osti.gov/biblio/392720
+
+    The function is suitable for spectra with asymmetric peaks, such as those
+    from CZT crystals. The exponential tails are characterized by 3 parameters: a
+    tail-to-peak amplitude ratio, a slope, and a cutoff parameter.
+
+    Parameters
+    ----------
+    x : numpy.ndarray
+        X-data
+    amp : float
+        Amplitude parameter
+    mu: float
+        Mean parameter
+    sigma: float
+        Width parameter
+    ltail_ratio: float
+        Left-side exponential tail amplitude / Amplitude
+    ltail_slope: float
+        Left-side exponential tail slope, affects "steepness" of the tail
+    ltail_cutoff: float
+        Left-side exponential tail cutoff, affects "length" of the tail
+    rtail_ratio: float
+        Right-side exponential tail amplitude / Amplitude
+    rtail_slope: float
+        Right-side exponential tail slope, affects "steepness" of the tail
+    rtail_cutoff: float
+        Right-side exponential tail cutoff, affects "length" of the tail
+
+    Returns
+    -------
+    numpy.ndarray
+        Y-data
+    """
+    alpha = -1.0 / (2.0 * sigma**2)
+    ltail_func = np.zeros(shape=(len(x),))
+    rtail_func = np.zeros(shape=(len(x),))
```

Comment:
You could do `np.zeros_like(x)` here

Reply:
`zeros_like` is [a lot slower](https://github.com/numpy/numpy/issues/9909), and this is in a critical place, so I think we should stick with `zeros`

Reply:
@micahfolsom what the ... 🤦 ... nice find.